syntax = "proto3";

package fileprocessor;

// Service definition
service FileProcessor {
  rpc ProcessFile(FileUploadRequest) returns (FileProcessResponse);
}

// Message for file upload request
message FileUploadRequest {
  string userId = 1;
  string fileId = 2;
  string filename = 3;
  bytes file = 4; // PDF file content
}

// Metadata associated with the file
message FileMetadata {
  string title = 1;
  int64 pageCount = 2;
  int64 filesize = 3;
  string locale = 4;
}

// Response message including processed file information
message FileProcessResponse {
  string fileId = 1;
  FileMetadata metadata = 2;
  repeated Page pages = 3; // Processed pages from AWS Textract
}

// Processed page content from the file
message Page {
  int64 pageId = 1;
  string content = 2; // Extracted text content of the page
}